README.md

```diff
@@ -3,237 +3,59 @@
-we must use the [Closure Compiler](https://developers.google.com/closure/compiler/) in
-either
-[SIMPLE or ADVANCED](https://developers.google.com/closure/compiler/docs/compilation_levels)
-mode with the **`--process_closure_primitives`** flag enabled (it is enabled by
-default). When enabled, if it finds a call to **`goog.setCssNameMapping()`** in
-any of its inputs, it will use the argument to `goog.setCssNameMapping()` as the
-basis of a renaming map that is applied at compile time. To create the
-appropriate renaming map with postcss-rename, use **`CLOSURE_COMPILED`** as
-the argument to **`outputRenamingMapFormat`**:
-
-```javascript
-module.exports = {
-  plugins: [
-    require('postcss-rename')({
-      outputRenamingMapFormat: 'CLOSURE_COMPILED',
-      renamingType: 'CLOSURE',
-      outputRenamingMap: 'renaming_map.js',
-    }),
-  ],
-};
-```
-
-This yields the following content for **`renaming_map.js`**:
-
-```javascript
-goog.setCssNameMapping({
-  "dialog": "a",
-  "content": "b",
-  "title": "c"
-});
-```
-
-Now **`renaming_map.js`** is a suitable input for the Closure Compiler. Recall
-our original snippet of JavaScript code:
-
-```javascript
-var element = goog.dom.getElementByClass(goog.getCssName('dialog-content'));
-```
-
-If passed to the Closure Compiler in SIMPLE mode along with
-**`renaming_map.js`**, it will be transformed to the following after
-compilation:
-
-```javascript
-var element = goog.dom.getElementByClass("a-b");
-```
-
-This achieves the goal of reducing both CSS and JS file sizes without changing
-the behavior of the application.
-
-Admittedly, using CSS renaming is a fairly advanced option that requires a
-well-organized build system to ensure that the appropriate CSS and JS assets are
-produced for both development and production. See
-[MoreOnCssRenaming](https://github.com/google/closure-stylesheets/wiki/More-on-CSS-Renaming)
-for more details on this topic.
-
-**Note:** it is also possible to exclude certain class names from being renamed
-by using the **`excludedClassesFromRenaming`** option. This may be necessary
-if some of your HTML is generated by a process that does not take CSS renaming
-into account. For example, if you are using a Python Django server and are using
-its template system, then any CSS classes used in those templates will not be
-renamed (unless you introduce a process to do so). In order to ensure that the
-JS and CSS that use the HTML reference CSS classes consistently, each CSS class
-in the Django template should be passed as an argument to postcss-rename
-with the **`excludedClassesFromRenaming`** option when generating the CSS.
-
-References to CSS class names that are excluded from renaming should _never_ be
-wrapped in `goog.getCssName()`, or else they run the risk of being partially
+users. It's designed to be used along with a plugin that can rewrite HTML
+templates and/or references in JS. If you write such a plugin, let us know and
+we'll link it here!
```

Comment:
> ... if people already have the closure compiler set up, they're probably already using it for their class renaming as well

I'm a bit stuck on what "plugin" means in this scenario, given that this wording is replacing the references to Closure Compiler and Closure Templates.

From my understanding the reference to Closure Compiler in the old readme also relies on getCssName in https://google.github.io/closure-library/api/goog.html, which isn't part of the compiler but instead the Closure Library. And that's what Closure Templates uses as well.

To my knowledge, these don't rewrite HTML templates and/or JS references if used without an optimizing compiler (such as Closure Compiler). Is something that performs such an HTML/JS optimizing step an example "plugin" in this scenario?

Do we envision a callback passed to `outputMapCallback` that outputs a mapping compatible with Closure Library/Closure Templates, for example? What's the plugin in this case?

Also a nit: "plugin" seems ambiguous in this context, because `postcss-rename` itself is a plugin.